pmm/commands.go

```diff
@@ -0,0 +1,1401 @@
+/*
+	Copyright (c) 2016, Percona LLC and/or its affiliates. All rights reserved.
+
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>
+*/
+
+package pmm
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	consul "github.com/hashicorp/consul/api"
+	"github.com/percona/pmm/proto"
+	"github.com/percona/pmm/proto/config"
+	"github.com/prometheus/client_golang/api/prometheus"
+	"github.com/roman-vynar/service"
+	"golang.org/x/net/context"
+	"gopkg.in/yaml.v2"
+)
+
+// PMM client config structure.
+type Config struct {
+	ServerAddress string `yaml:"server_address"`
+	ClientAddress string `yaml:"client_address"`
+	ClientName    string `yaml:"client_name"`
+}
+
+// Service status description.
+type instanceStatus struct {
+	Type    string
+	Name    string
+	Port    string
+	Status  string
+	DSN     string
+	Options string
+}
+
+// Definition of program for service manager.
+type program struct{}
+
+func (p *program) Start(s service.Service) error {
+	return nil
+}
+
+func (p *program) Stop(s service.Service) error {
+	return nil
+}
+
+func (p *program) run() error {
+	return nil
+}
+
+// Main class.
+type Admin struct {
+	ServiceName string
+	ServicePort uint
+	Config      *Config
+	filename    string
+	qanapi      *API
+	consulapi   *consul.Client
+}
+
+// LoadConfig read PMM client config file.
+func (a *Admin) LoadConfig(filename string) error {
+	a.filename = filename
```

Comment:
This can be written as
```
    a.filename = filename
	a.Config = &Config{}
	if !FileExists(filename) {
	    return nil
 	}
 	bytes, err := ioutil.ReadFile(filename)
 	if err != nil {
 		return err
 	}
 	if err := yaml.Unmarshal(bytes, config); err != nil {
 		return err
 	}
 	return nil
```